src/main/java/org/commonwl/view/cwl/CWLService.java

```diff
@@ -204,9 +206,11 @@ public Workflow parseWorkflowNative(
       }
       if (!found && !packedWorkflowId.isEmpty()) throw new WorkflowNotFoundException();
     }
-    if (!found && extractProcess(cwlFile) == CWLProcess.WORKFLOW) {
-      // Check the current json node is a workflow
+    if (!found && (extractProcess(cwlFile) == CWLProcess.WORKFLOW)) {
       found = true;
+    } else if (found && (extractProcess(cwlFile) != CWLProcess.WORKFLOW)) {
```

Comment:
Interesting, @mr-c. I tried writing a unit test for this, but I couldn't make the service to reach this line. The `found` starts as `false`. And unless it's set before, it will not enter this `else if` block.

To make it eval to `true`, I packed the CommandLineTool,

```cwl
cwlVersion: v1.2
$graph:
- id: echo
  class: CommandLineTool
  baseCommand: echo
  inputs: []
  outputs: []
```

but that still results in `false` since the code above already checks if it is a packed workflow, and only sets `found` to `true` in that case.

Did you reach this part of the code testing with the application running?

Reply:
I tried with

```cwl
cwlVersion: v1.2

# What type of CWL process we have in this document.
class: CommandLineTool

label: Process DL0 to DL3

doc: >
    This tool (must become a workflow) 
    processes DL0 to DL3 simulated data.

baseCommand: echo

inputs:
    metadata_queue:
        type: File
        inputBinding: 
            prefix: --metadata-queue
            position: 1
    
outputs:
    dl3:
        type: File[]
        outputBinding:
            glob: '*.txt'
    dl3_metadata:
        type: stdout
    irfs:
        type: File[]
        outputBinding:
            glob: '*.txt'
    irfs_metadata:
        type: stdout

stdout: output.txt
```

And it still couldn't reach that part of the code. Here's the test:

```diff
diff --git a/src/test/java/org/commonwl/view/cwl/CWLServiceTest.java b/src/test/java/org/commonwl/view/cwl/CWLServiceTest.java
index 384fcd4..fe6993d 100644
--- a/src/test/java/org/commonwl/view/cwl/CWLServiceTest.java
+++ b/src/test/java/org/commonwl/view/cwl/CWLServiceTest.java
@@ -442,4 +442,24 @@ public class CWLServiceTest {
       assertEquals(CWLProcess.COMMANDLINETOOL, steps.get("lobSTR").getRunType());
     }
   }
+
+  @Test
+  public void testExceptionRaisedForCommandLineTool() {
+    Exception thrown =
+            Assertions.assertThrows(
+                    CWLNotAWorkflowException.class,
+                    () -> {
+                      CWLService cwlService =
+                              new CWLService(
+                                      Mockito.mock(RDFService.class),
+                                      Mockito.mock(CWLTool.class),
+                                      Mockito.mock(GitConfig.class).licenseVocab(),
+                                      999999);
+                      cwlService.parseWorkflowNative(
+                              Paths.get("src/test/resources/cwl/invalid_is_a_packed_command_line.cwl"), "");
+                    });
+    assertEquals(
+            "File 'lobSTR-workflow.cwl' is over singleFileSizeLimit - 2 KB/0 bytes",
+            thrown.getMessage());
+  }
 }
```

Just create the CWL file at `src/test/resources/cwl/invalid_is_a_packed_command_line.cwl`

Reply:
(WIP test, I copied another test and was modifying it, slowly altering and re-running it)